doc/architectural/central-data-structures.md

```diff
@@ -0,0 +1,91 @@
+# Central Data Structures
+
+The following is some light technical documentation of the major data structures
+represented in the input transformation to Intermediate Representation (IR) inside
+CBMC and the assorted CProver tools.
+
+## goto_modelt
+
+The `goto_modelt` is the main data structure that CBMC (and the other tools) use to
+represent GOTO-IR (the `GOTO` Intermediate Representation).
+
+A `goto_modelt` is effectively a product type, consisting of:
```

Comment:
Same comment as on the previous draft; I would suggest using "pair" or "tuple" as a more C++-ish way of saying this.